shell/platform/windows/text_input_plugin.h

```diff
@@ -20,33 +20,52 @@
 
 namespace flutter {
 
-class TextInputPluginDelegate;
+class FlutterWindowsEngine;
 
 // Implements a text input plugin.
 //
 // Specifically handles window events within windows.
 class TextInputPlugin {
  public:
-  explicit TextInputPlugin(flutter::BinaryMessenger* messenger,
-                           TextInputPluginDelegate* delegate);
+  TextInputPlugin(flutter::BinaryMessenger* messenger,
+                  FlutterWindowsEngine* engine);
 
   virtual ~TextInputPlugin();
 
+  // Called when the Flutter engine receives a raw keyboard message.
   virtual void KeyboardHook(int key,
                             int scancode,
                             int action,
                             char32_t character,
                             bool extended,
                             bool was_down);
 
+  // Called when the Flutter engine receives a keyboard character.
   virtual void TextHook(const std::u16string& text);
 
+  // Called on an IME compose begin event.
+  //
+  // Triggered when the user begins editing composing text using a multi-step
+  // input method such as in CJK text input.
   virtual void ComposeBeginHook();
 
+  // Called on an IME compose commit event.
+  //
+  // Triggered when the user commits the current composing text while using a
+  // multi-step input method such as in CJK text input. Composing continues with
+  // the next keypress.
```

Comment:
uber-nit: it's not necessarily the user who explicitly performs a commit -- in Korean, for example, just typing the next character is enough to trigger a commit of the composing region. Maybe "Called when the user triggers a commit of the [...]" to be very slightly more accurate.

Reply:
Thanks for the additional info, I'm not very familiar with this area. Updated! :)